### PR TITLE
Increased user experience

### PR DESCRIPTION
**Pull request by arimet**
- [x] Remonter les boutons "nouveau champ" au même niveau que les tabs (supprimer les espaces rouges) 
 - [x] Nouveau champ depuis une colonne : mettre la datagrid dans une modal, même taille que la modale paramétrage d'un champ, titre de la modale : "Sélectionner une colonne"
  - [x] Fermer modale champ quand click sur backdrop
 - [x]  Faire que la zone de drag n drop des champs soit responsive (prenne 100% de l'espace disponible)
- [x] Quand on vient de créer un champ, scroll au niveau de ce champ
- [x] Sur l'import d'un modèle, ne pas mettre le warning si le modèle actuel est vide

  

- [x]  Enrichment :
    - mieux aligner les champs Source column et sub path
    - Aérer un peu le titre, la carte, et le bouton save
    - Revoir couleur "Mode avancé"

### CODE REVIEW

File: src/app/js/admin/Appbar/ClearDialog.js [hunk 1]
```diff
@@ -38,6 +38,9 @@ const ClearDialogComponent = props => {
 
     useEffect(() => {
         if (succeeded) {
+            window.location.replace(
```
Comment (floo51) [inline src/app/js/admin/Appbar/ClearDialog.js]: You sure we want that? It'll trigger a full page reload, not a react-router navigation

File: src/app/js/admin/Appbar/AddFieldFromColumnButton.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ export const AddFieldFromColumnButtonComponent = ({
 AddFieldFromColumnButtonComponent.propTypes = {
     onShowExistingColumns: PropTypes.func.isRequired,
     p: polyglotPropTypes.isRequired,
-    scope: PropTypes.string.isRequired,
```
Comment (floo51) [inline src/app/js/admin/Appbar/AddFieldFromColumnButton.js:48]: Is this related to the change?

File: src/app/js/admin/ImportModelButton.js [hunk 3]
```diff
@@ -22,31 +27,59 @@ const useStyles = makeStyles({
...
-    const [open, setOpen] = useState(false);
     const [
         showImportFieldsConfirmation,
         setShowImportFieldsConfirmation,
     ] = useState(false);
-    const [showSuccessAlert, setShowSuccessAlert] = useState(false);
 
-    const handleImportFieldsClose = sucess => {
+    const [showAlert, setShowAlert] = useState(false);
+    const [useInput, setUseInput] = useState(false);
```
Comment (floo51) [inline src/app/js/admin/ImportModelButton.js]: Careful with this kind of naming in react, we could think it's a hook (and not the result of a hook)

File: src/app/js/admin/ImportModelButton.js [hunk 4]
```diff
@@ -22,31 +27,59 @@ const useStyles = makeStyles({
...
     const classes = useStyles();
-    const [open, setOpen] = useState(false);
     const [
         showImportFieldsConfirmation,
         setShowImportFieldsConfirmation,
     ] = useState(false);
-    const [showSuccessAlert, setShowSuccessAlert] = useState(false);
 
-    const handleImportFieldsClose = sucess => {
+    const [showAlert, setShowAlert] = useState(false);
```
Comment (floo51) [inline src/app/js/admin/ImportModelButton.js]: showAlert seems like a function, because it has a verb, but it's a status

File: src/app/js/admin/subresource/DeleteSubresourceButton.spec.js [hunk 5]
```diff
@@ -57,7 +57,7 @@ describe('<DeleteSubresourceButton />', () => {
         );
 
         const okButton = wrapper.findWhere(node => {
-            return node.type() === Button && node.text() === 'OK';
+            return node.type() === Button && node.text() === 'Accept';
```
Comment (floo51) [inline src/app/js/admin/subresource/DeleteSubresourceButton.spec.js:60]: We sure about this? What about translations?

File: src/app/js/fields/FieldGrid.js [hunk 6]
```diff
@@ -199,10 +207,30 @@ const DraggableItemGrid = ({
...
+
     useEffect(() => {
         setItems(buildFieldsDefinitionsArray(fields));
     }, [JSON.stringify(fields)]);
 
+    const fieldsLength = fields.length;
+    const previousFieldsLength = useRef(fields.length);
+    useEffect(() => {
+        if (previousFieldsLength.current + 1 === fields.length) {
+            setTimeout(() => {
```
Comment (floo51) [inline src/app/js/fields/FieldGrid.js]: This could be in a separate well-named function to make it clearer

File: src/app/js/fields/FieldGrid.js [hunk 7]
```diff
@@ -88,6 +93,9 @@ const useStyles = makeStyles({
     },
 });
 
+// 1rem = 16px
+const ROOT_PADDING = 16;
```
Comment (floo51) [inline src/app/js/fields/FieldGrid.js]: Maybe we could take advantage of this to use the same variable in the styles.
Also, 1rem != 16px, it's relative to the font-size, font-size change = padding change

File: src/app/js/fields/wizard/SelectDatasetField.js [hunk 8]
```diff
@@ -34,7 +34,7 @@ export const SelectDatasetFieldComponent = ({
             {datasetFields.map(datasetField => (
                 <MenuItem
                     key={`id_${datasetField}`}
-                    className={`column-${datasetField}`}
+                    className={`column-${datasetField.replaceAll(' ', '-')}`}
```
Comment (floo51) [inline src/app/js/fields/wizard/SelectDatasetField.js:37]: What if there are other characters like ' " \ ... ?

File: src/app/js/public/resource/Resource.js [hunk 9]
```diff
@@ -85,7 +85,8 @@ export class ResourceComponent extends React.Component {
         if (
             match.params &&
             match.params.uri &&
-            match.params.uri.length !== 32 && // md5 length for subresources uuid
+            match.params.uri.length !== 32 &&
+            !match.params.uri.includes('%2F') && // md5 length for subresources uuid
```
Comment (floo51) [inline src/app/js/public/resource/Resource.js:89]: Why not allowing a html encoded slash?

**Review by floo51 — CHANGES_REQUESTED** [review]

File: src/app/js/admin/Appbar/AddFieldFromColumnButton.js (hunk 2 above)
Comment (arimet) [reply src/app/js/admin/Appbar/AddFieldFromColumnButton.js:48]: No, but it was an error that went up in the console. So I took the opportunity to correct it

File: src/app/js/admin/subresource/DeleteSubresourceButton.spec.js (hunk 5 above)
Comment (arimet) [reply src/app/js/admin/subresource/DeleteSubresourceButton.spec.js:60]: So far I haven't seen any translations in the tests (maybe I missed them). It would be interesting to put them in place but not the subject of the ticket in my opinion

File: src/app/js/public/resource/Resource.js (hunk 9 above)
Comment (arimet) [reply src/app/js/public/resource/Resource.js:89]: html encoded slash is the old version of sub-resources. Because Julien made the change, we didn't think that it would affect the old models. So adding this condition allows us to check both possible cases.
